delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/CompositeViewFactorySuite.scala

```diff
@@ -0,0 +1,291 @@
+import ch.epfl.bluebrain.nexus.delta.rdf.query.SparqlQuery.SparqlConstructQuery
+import ch.epfl.bluebrain.nexus.delta.rdf.syntax.iriStringContextSyntax
+import ch.epfl.bluebrain.nexus.delta.sdk.projects.model.ProjectBase
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Tag.UserTag
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{Identity, Label, ProjectRef}
+import ch.epfl.bluebrain.nexus.testkit.bio.BioSuite
+import io.circe.{Json, JsonObject}
+
+import java.util.UUID
+
+class CompositeViewFactorySuite extends BioSuite {
+
+  implicit private val projectBase: ProjectBase = ProjectBase.unsafe(iri"http://localhost/project")
+  private val uuid                              = UUID.randomUUID()
+  implicit private val uuidF: UUIDF             = UUIDF.fixed(uuid)
+
+  private val schemas: Set[IriOrBNode.Iri] = Set(nxv + "Schema")
+  private val types: Set[IriOrBNode.Iri]   = Set(nxv + "Type")
+  private val tag: Some[UserTag]           = Some(UserTag.unsafe("tag"))
+  private val includeDeprecated            = true
+  private val includeMetadata              = true
+  private val includeContext               = true
+
+  private val projectSourceId     = iri"http://localhost/project-source"
+  private val projectSourceFields = ProjectSourceFields(
+    Some(projectSourceId),
+    schemas,
+    types,
+    tag,
+    includeDeprecated
+  )
+
+  private val crossSourceId     = iri"http://localhost/cross-project-source"
+  private val crossSourceFields = CrossProjectSourceFields(
+    Some(crossSourceId),
+    ProjectRef(Label.unsafe("org"), Label.unsafe("otherproject")),
+    identities = Set(Identity.Anonymous),
+    schemas,
+    types,
+    tag,
+    includeDeprecated
+  )
+
+  private val remoteSourceId     = iri"http://localhost/remote-project-source"
+  private val remoteSourceFields = RemoteProjectSourceFields(
+    Some(remoteSourceId),
+    ProjectRef.unsafe("org", "remoteproject"),
+    Uri("http://example.com/remote-endpoint"),
+    Some(Secret("secret token")),
+    schemas,
+    types,
+    tag,
+    includeDeprecated
+  )
+
+  private val esProjectionId     = iri"http://localhost/es-projection"
+  private val esProjectionFields = ElasticSearchProjectionFields(
+    Some(esProjectionId),
+    SparqlConstructQuery.unsafe("CONSTRUCT..."),
+    None,
+    JsonObject("mapping" -> Json.obj()),
+    ContextObject(JsonObject("context" -> Json.obj())),
+    Some(JsonObject("settings" -> Json.obj())),
+    schemas,
+    types,
+    tag,
+    includeDeprecated,
+    includeMetadata,
+    includeContext
+  )
+
+  private val blazegraphProjectionId     = iri"http://example.com/blazegraph-projection"
+  private val blazegraphProjectionFields = SparqlProjectionFields(
+    Some(blazegraphProjectionId),
+    SparqlConstructQuery.unsafe("CONSTRUCT..."),
+    schemas,
+    types,
+    tag,
+    includeDeprecated,
+    includeMetadata
+  )
+
+  test("Create the matching source from the project source field with a defined id") {
```

Comment:
What is a matching source?

Reply:
More broadly I don't really understand what these tests are actually asserting. What is the API doing?

Reply:
That the fields are correctly transformed into a value